KitbotExampleWalkthrough.md

```diff
@@ -0,0 +1,245 @@
+      this.setVoltages(speeds.left * 12, speeds.right * 12);
+    }, this);
+}
+```
+
+This method is named `setVoltagesArcadeCommand` after the arcade drive inverse kinematics (IK) it uses, but other types of IK can also work.
+On a real robot what we use would come down to driver preference.
+In the method itself we have a very similar structure to the previous method.
+The main difference is that instead of having one method in our lambda (`() ->`), we have a code block ({} block).
+This means we need to end each line with a ; but can use multiple lines in our Command.
+In this case we call arcadeDriveIK from WPILib's `DifferentialDrive` class and store the result in a variable called `speeds`.
+Then we pass speeds to the drive method, multiplying by 12 to convert from the -1..1 units from the IK method to voltage.
+
+These methods are all well and good on their own, but it would be nice if we had a way to actually use them.
+Lets bind them to a joystick.
+Go to RobotContainer.java and add a `CommandXboxController` object.
+
+```Java
+CommandXboxController controller = new CommandXboxController(0);
+```
+
+`CommandXboxController` is a convenience wrapper around a `XboxController` that makes it easy to bind commands to buttons on the controller.
+We also need to add an instance of the `DrivetrainSubsystem` to RobotContainer so that we can control it.
+
+```Java
+CommandXboxController controller = new CommandXboxController(0);
+
+DrivetrainSubsystem drivetrainSubsystem = new DrivetrainSubsystem();
+```
+
+Finally we can bind the arcade drive command to the joysticks.
+To do this call `setDefaultCommand()` on `drivetrainSubsystem` in RobotContainer's `configureBindings()` method.
+This sets a Command that will be run when no other command is using the drivetrain.
+Pass in `drivetrainSubsystem.setVoltagesArcadeCommand()` as the default Command.
+
+For the `DoubleSuppliers` to the arcade drive command we can add two lambdas that call `getLeftY()` and `getRightX()` on the controller, which looks at the current joysick values.
+This means that each loop this command is run, it will check the current joystick values and turn them into voltages for the wheels.
+
+This line is a little long now, so break it up into a few lines to keep it readable.
+
+```Java
+drivetrainSubsystem.setDefaultCommand(
+    drivetrainSubsystem.setVoltagesArcadeCommand(
+        () -> controller.getLeftY(), 
+        () -> controller.getRightX()));
+```
+
+Using the joystick values directly is perfectly fine, but sometimes our driver wants a different feel to the controls.
+One common way is to square the inputs to the drivetrain, which reduces the sensitivity of the drivebase at low speeds.
+Another is to add a "deadband" where we wont output any voltage if the input is less than some threshold.
+This stops the robot from moving if the controllers don't perfectly read 0 when they aren't pressed.
+Lets add a function in RobotContainer to modify the joystick inputs.
```

Comment:
When I explain this I find that drawing the graph often helps a lot. You might want a picture here.